lib/executor/ramping_arrival_rate.go

```diff
@@ -415,12 +435,15 @@ func (varr RampingArrivalRate) Run(ctx context.Context, out chan<- stats.SampleC
 				varr.logger.Warningf("Insufficient VUs, reached %d active VUs and cannot allocate more", maxVUs)
 				continue
 			}
-			initVU, err := varr.executionState.GetUnplannedVU(maxDurationCtx, varr.logger)
-			if err != nil {
-				return err
+
+			select {
+			case makeUnplannedVUCh <- struct{}{}:
+				// this is the only goroutine that touches remainingUnplannedVUs and if we didn't
+				// send on the channel no new unplannedVU will be stared so no need to decrease it
+				remainingUnplannedVUs--
+				vu = <-activeVUs // just get any VU that gets activated, whether it is the unplanned or not doesn't matter
```

Comment:
I don't think this is the right approach to initializing VUs mid-test. This would still make the executor stop and wait for the VU to initialize or another VU to be returned. It could horribly skew and time-shift the arrival rate...

Instead, I think the correct approach is "No VU?! :( Oh well, we miss this iteration, but we'll try to execute the next one, while maybe allocating a VU in the background, if there are some unplanned budget remaining". That is, something non-blocking like this:
- if we have to make an iteration but there's no VU, emit a `missed_iteration` metric (or add a `TODO` for now, we can discuss the actual metric specifics tomorrow)
- if there are `remainingUnplannedVUs`, signal through `makeUnplannedVUCh` to start initializing a new one
- it's important for that "signaling" to occur in a `select{}` with a `default` case, so it doesn't block if we're already in the process of initializing an unplanned VU from a previous iteration - we shouldn't block, but we should also never initialize more than 1 VU at a time

That is, if the schedule calls for us to make an iteration at time X, and we don't have a free VU at that time, this iteration will forever be considered missed and we won't try to get it back. Instead, we'll move on to the next one, while potentially increasing VUs in the hopes of eventually having enough to execute every iteration.

Reply:
Resolved by https://github.com/loadimpact/k6/pull/1500/commits/502a550c7a7328f04f6810dceefb8f7b3decb8c8